code/game/objects/items/explosives/mine.dm

```diff
@@ -1,277 +1,1277 @@
+	var/current_fuel = 0
+	///The maximum capacity of fuel units
+	var/max_fuel = 3
+	///Time between pulses
+	var/time_between_pulses = 10 SECONDS
+	///Stored reference to the timer that determines when extra_effects() is called again
+	var/pulse_timer
+
+	//Light-related vars for when the radiation glow is emitted
+	light_power = 10
+	light_color = COLOR_GREEN
+	light_system = HYBRID_LIGHT
+	light_mask_type = /atom/movable/lighting_mask/flicker
+
+/obj/item/mine/radiation/examine(mob/user)
+	. = ..()
+	. += "Currently has [span_bold("[current_fuel]")] out of [span_bold("[max_fuel]")] fuel."
+
+/obj/item/mine/radiation/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	//While we could use a var/fuel_type instead of hard coded to use uranium, it would be a headache since only /stack/ use amount vars
+	//Also no other /stack/ type apart from maybe phoron would be a good fuel candidate so why bother
+	if(istype(I, /obj/item/stack/sheet/mineral/uranium))
+		if(current_fuel >= max_fuel)
+			balloon_alert(user, "Already full!")
+			return
+
+		var/obj/item/stack/uranium = I
+		var/amount_to_transfer = max(1, min(uranium.amount, max_fuel - current_fuel))
+		if(!uranium.use(amount_to_transfer))
+			balloon_alert(user, "Not enough fuel!")
+			return
+
+		current_fuel += amount_to_transfer
+		balloon_alert(user, current_fuel == max_fuel ? "Full!" : "[current_fuel]/[max_fuel] fuel!")
+		return
+
+/obj/item/mine/radiation/setup(mob/living/user)
+	if(!current_fuel)
+		balloon_alert(user, "No fuel!")
+		return
+	. = ..()
+
+/obj/item/mine/radiation/disarm()
+	//if(pulse_timer)
+	deltimer(pulse_timer)
+	light_range = 0	//Hybrid lights don't actually turn off, just have to change light_range and update
+	update_light()
+	return ..()
+
+/obj/item/mine/radiation/trip_mine(atom/movable/victim)
+	if(!current_fuel)
+		return FALSE
+	return ..()
+
+/obj/item/mine/radiation/extra_effects(atom/movable/victim)
+	if(!current_fuel)
+		disarm()
+		return
+
+	current_fuel--
+	new /obj/effect/temp_visual/shockwave(get_turf(src), radiation_range * 1.5)
+	light_range = radiation_range * 1.5
+	set_light(light_range, light_power, light_color)
+
+	var/list/exclusion_zone = circle_range(get_turf(src), radiation_range)	//Radiation SHALL NOT pass through walls (too much cope)
+	for(var/mob/living/carbon/radiation_victim in exclusion_zone)
+		if(!check_path(src, radiation_victim, FALSE, TRUE, TRUE, TRUE))	//circle_range() goes through walls, so make sure the target is reachable
+			exclusion_zone -= radiation_victim	//Remove it since it will be iterating again below
+			continue
+
+		//Apply initial damages of the detonation evenly in BURN and TOX, then do a fifth of it in cellular damage
+		radiation_victim.apply_damages(0, radiation_damage/2, radiation_damage/2, 0, radiation_damage/5, ishuman(radiation_victim) ? pick(GLOB.human_body_parts) : null, BIO)
+		radiation_victim.adjust_stagger(radiation_damage/5 SECONDS)
+		radiation_victim.adjust_radiation(radiation_damage SECONDS)
```

Comment:
bro why is this so aids? The rad effect does damage itself, but you're stacking it with more damage, and there's no dead check so it will keep spamming corpses. The way you've made this, a rad mine can be like 6 times stronger than a rad grenade. In what world is this a good idea?

Reply:
Just don't stand in the field? I can make the damages lower but all I see is a skill issue. Design of the mine is to create a hazard no one should cross.

I can add a dead check I just dislike dying making you invulnerable. 🤷